Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -504,17 +728,86 @@ public ArraySegment<byte> GetPayloadBytes(FileAccessSetup setup, MemoryStream st
                 WriteInjectionTimeoutBlock(writer, timeoutMins);
                 WriteTranslationPathStrings(writer, DirectoryTranslator);
                 WriteErrorDumpLocation(writer, InternalDetoursErrorNotificationFile);
-                WriteFlagsBlock(writer);
+                WriteFlagsBlock(writer, m_fileAccessManifestFlag);
                 WriteExtraFlagsBlock(writer, FileAccessManifestExtraFlag.None);
                 WritePipId(writer, PipId);
                 WriteReportBlock(writer, setup);
                 WriteDllBlock(writer, setup);
-                WriteManifestTreeBlock(writer);
+                if (!IsManifestTreeBlockSealed)
+                {
+                    WriteManifestTreeBlock(writer);
+                }
+                else
+                {
+                    Contract.Assert(m_sealedManifestTreeBlock != null);
+                    writer.Write(m_sealedManifestTreeBlock);

```

Comment:
are we going to have to update MacSandbox to support `m_sealedManifestTreeBlock`?  (i.e., is that ever going to be used on macOS?)

Reply:
:+1"

---
In reply to: [276463940](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276463940) [](ancestors = 276463940)

Reply:
I don't know if we're ever going to use it on MacOS. But for sure, to use it on Mac, we need to ensure that the manifest tree is not modified once we create `ISandboxedProcess`. Currently it's not the case for Mac. are we going to have to update MacSandbox to support `m_sealedManifestTreeBlock`?  

---
In reply to: [276463940](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276463940) [](ancestors = 276463940)